### PR TITLE
feat: Run all doctor checks & print summary

### PR DESCRIPTION
**Pull request by acburdine**

### CODE REVIEW

File: lib/commands/doctor/checks/system-stack.js [hunk 1]
```diff
@@ -35,24 +37,35 @@ function systemStack(ctx) {
...
+            Promise.resolve({yes: false})
+        ).then((answer) => {
+            if (answer.yes) {
+                task.skip('System stack check skipped');
+                return Promise.resolve();
+            }
+
+            return Promise.reject(new errors.SystemError({
+                message: `System stack checks failed with message: '${error.message}'`,
+                task: 'Checking operating system compatibility'
```
Comment (kirrg001) [inline lib/commands/doctor/checks/system-stack.js]: I guess you can use `taskTitle` here.

Comment (acburdine) [reply lib/commands/doctor/checks/system-stack.js]: oops 🙈

File: lib/commands/doctor/checks/system-stack.js [hunk 2]
```diff
@@ -35,24 +37,35 @@ function systemStack(ctx) {
...
+    });
 }
 
 module.exports = {
-    title: 'Checking operating system compatibility',
+    title: taskTitle,
     task: systemStack,
     enabled: (ctx) => !ctx.local,
-    skip: (ctx) => ctx.argv && !ctx.argv.stack,
+    skip: (ctx) => !ctx.isDoctorCommand && ctx.argv && !ctx.argv.stack,
```
Comment (kirrg001) [inline lib/commands/doctor/checks/system-stack.js]: What is the diff between `ctx.local` and `argv.stack`? 🤔 
Aren't they the same somehow? `argv.stack` is false if local PM is used.

Comment (kirrg001) [reply lib/commands/doctor/checks/system-stack.js]: Discussed.

`--no-stack` => `argv.stack=false` (but can be local PM or a different extension)

**Pull request merged**